org-roam-db.el

```diff
@@ -133,7 +133,8 @@ SQL can be either the emacsql vector representation, or a string."
 
     (refs
      [(ref :unique :not-null)
-      (file :not-null)])))
+      (file :not-null)
+      (key :not-null)])))
```

Comment:
Let's call this `type` instead, same as the `links` table. Key will then take on values:

1. `cite` if the prefix is an org-ref cite link (as per `org-ref-cite-types`)
2. `website` if it's a URL (optional, nothing is being done with this)
3. `roam' otherwise.

Reply:
Wouldn't this break setups where the `ROAM_KEY` is just the citekey?

e.g. `ROAM_KEY: ahrens2017` would be detected as a roam type rather than a cite type.

having the `key` (or whatever column name is chosen) just be the original `ROAM_KEY` stripped of any `org-ref-cite-types` prefixes should allow both `cite:ahrens2017` and `ahrens2017` to work

Reply:
I think it's fine to keep the `cite:` in `ROAM_KEY` as a way to identify bibliographic notes.  It comes at very little cost to us, and I agree with @jethrokuan that it helps with entry if you're not using `org-roam-bibtex` to generate those files.

This stems from the desire to have different types of keys co-exists with just one file-metadata tag.  Otherwise, another option which wouldn't be as elegant would be to have different file-metadata tags like `ROAM_CITEKEY`, `ROAM_URLKEY`, etc.

Reply:
Fair enough. I can see the logic for future extendability. `ROAM_CITEKEY` and `ROAM_URLKEY` does not sound like a nice solution IMO.

I might submit a quick PR to fix the backlinks buffer not catering for all cite types